pkg/cmd/grafana-cli/commands/install_command.go

```diff
@@ -97,7 +97,9 @@ func InstallPlugin(pluginName, version string, c utils.CommandLine, client utils
 
 		// Plugins which are downloaded just as sourcecode zipball from github do not have checksum
 		if v.Arch != nil {
-			checksum = v.Arch[osAndArchString()].SHA256
+			if configuredChecksum := v.Arch[osAndArchString()].SHA256; len(configuredChecksum) > 0 {
```

Comment:
I think that you have to check whether `v.Arch` map contains a key for the specific architecture and if not then to fall back to the sha256 checksum of the `any` key.
As an example of a plugin that has architecture specific checksums you can use [Azure Data Explorer Datasource](https://grafana.com/api/plugins/grafana-azure-data-explorer-datasource) and for a plugin that has a generic checksum you can use [Zabbix plugin for Grafana](https://grafana.com/api/plugins/alexanderzobnin-zabbix-app).
So for testing it you can run:
`./bin/grafana-cli plugins install grafana-azure-data-explorer-datasource`
`./bin/grafana-cli plugins install alexanderzobnin-zabbix-app`
In the first case, your architecture specific checksum should be used (if you running the command in Windows should be the `windows-amd64`) and in the latter the `any` checksum should be used.

Reply:
Agree with @papagian, I made a suggestion in [another](https://github.com/grafana/grafana/pull/30035#discussion_r551438624) comment.

Reply:
checked with both plugin, code works correctly.